content/edpratti/collections/git-guide/posts/git-into-the-weeds/index.md

```diff
@@ -0,0 +1,240 @@
+
+## Soft
+
+<img src="./git_reset_soft.svg" alt="A diagram of the behavior of a Git reset with the soft parameter. It states that the HEAD moves to the specified commit, while the changes remain staged, and the working directory remains unchanged." style="background-color: var(--background_primary-bright); border: var(--border-width_s) solid var(--background_disabled); border-radius: var(--corner-radius_xl)">
+
+A soft reset maintains all changes but undoes a commit. This means that the working directory remains unchanged, but all changes are staged, and ready to be committed again.
+
+> **Within the context of a Git GUI, a soft Git reset would look like this::**
+
+<img src="./git_reset_soft_history.svg" alt="A Git commit history and a staging section that shows how the soft Git reset leaves changes staged while removing the commits." style="background-color: var(--background_primary-bright); border: var(--border-width_s) solid var(--background_disabled); border-radius: var(--corner-radius_xl)">
+
+## Mixed (Default)
+
+<img src="./git_reset_mixed.svg" alt="A diagram of the behavior of a Git reset with the mixed parameter. It states that the HEAD moves to the specified commit, the working directory maintains the changes, but they are unstaged." style="background-color: var(--background_primary-bright); border: var(--border-width_s) solid var(--background_disabled); border-radius: var(--corner-radius_xl)">
+
+A mixed reset undoes a commit and also unstages changes, but maintains edits in the working directory. This means you can select which changes you'd like to stage again.
+
+> **Within the context of a Git GUI, a soft Git reset would look like this::**
+
+<img src="./git_reset_mixed_history.svg" alt="A Git commit history and a staging section that shows how the default Git reset leaves changes unstaged while removing the commits." style="background-color: var(--background_primary-bright); border: var(--border-width_s) solid var(--background_disabled); border-radius: var(--corner-radius_xl)">
+
+## Hard
+
+<img src="./git_reset_hard.svg" alt="A diagram of the behavior of a Git reset with the hard parameter. It states that the HEAD moves to the specified commit, while all changes are deleted, and nothing is kept." style="background-color: var(--background_primary-bright); border: var(--border-width_s) solid var(--background_disabled); border-radius: var(--corner-radius_xl)">
+
+A hard reset, as the name implies, completely nullifies all changes and removes any trace of them from the repository. ***This cannot be undone, so it's incredibly important to be careful about when to use this command.***
+
+> **Within the context of a Git GUI, the commits would simply disappear.:**
+
+---
+
+# Git conflicts
+
+There are situations where a partiular change may cause conflicts. This can be triggered in different situations such as:
```

Comment:
Git conflicts actually modify text files by adding in:

````
<<<<<<< HEAD
Changes from the current branch
=======
Changes from the other branch
>>>>>>> branch-name
````

Do we wanna mention this there as well?